pymc/step_methods/hmc/quadpotential.py

```diff
@@ -94,7 +98,17 @@ def __str__(self):
 
 
 class QuadPotential:
-    def velocity(self, x, out=None):
+    dtype: np.dtype
+
+    @overload
+    def velocity(self, x: np.ndarray, out: None) -> np.ndarray:
```

Comment:
Shouldn't `out` get a real type here? Because `None` is not a type..

Reply:
You're technically correct. Although the singleton `None` has type `NoneType`, I believe that type checkers automatically do an implicit conversion here. (Who really wants to remember to type the extra letters and be so pedantic?)

If you check the mypy docs, they always use `None`, so it's not a problem.